sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingQueryListenerBus.scala

```diff
@@ -63,18 +83,29 @@ class StreamingQueryListenerBus(sparkListenerBus: LiveListenerBus)
     }
   }
 
+  /**
+   * Dispatch events to registered StreamingQueryListeners. Only the events associated queries
+   * started in the same SparkSession as this ListenerBus will be dispatched to the listeners.
+   */
   override protected def doPostEvent(
       listener: StreamingQueryListener,
       event: StreamingQueryListener.Event): Unit = {
+    val runIdsToReportTo = activeQueryRunIds.synchronized { activeQueryRunIds.toSet }
```

Comment:
Why need to clone the set? You can just use `activeQueryRunIds.synchronized { activeQueryRunIds.contains(...) }`. Right?